cmake/OpenCVFindProtobuf.cmake

```diff
@@ -67,22 +74,51 @@ else()
   endif()
 endif()
 
+# See https://github.com/opencv/opencv/issues/24369
+# In Protocol Buffers v22.0 and later drops C++11 support and depends abseil-cpp.
+#   Details: https://protobuf.dev/news/2022-08-03/
+# And if std::text_view is in abseil-cpp requests C++17 and later.
+
+if(HAVE_PROTOBUF)
+    if("${Protobuf_VERSION}" MATCHES [[[0-9]+.([0-9]+).[0-9]+]])
+        string(COMPARE GREATER "${CMAKE_MATCH_1}" "21" REQUEST_AFTER_CXX17)  # >=22
```

Comment:
What is the problem with `if'(NOT (Protobuf_VERSION VERSION_LESS 22))`?

https://cmake.org/cmake/help/latest/command/if.html#version-less

> Any non-integer version component or non-integer trailing part of a version component effectively truncates the string at that point.

Reply:
Yes, It is better than my script. Thank you very much !